src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -115,6 +118,7 @@ export class IndexPattern implements IIndexPattern {
     this.fieldAttrs = spec.fieldAttrs || {};
     this.intervalName = spec.intervalName;
     this.allowNoIndex = spec.allowNoIndex || false;
+    this.runtimeFieldMap = spec.runtimeFieldMap || {};
```

Comment:
why a separate runtimeFieldMap, instead of putting this on fieldsMap ?

Reply:
Its possible that a new type of runtime field would create more than one field on the index pattern.